### DISCUSSION TITLE
Store App definitions in git

### DISCUSSION THREAD

**Discussion by BeyerJC** [argoproj/argo-cd#10086] [Q&A] [answered]
Hey everyone!
I googled and searched examples but did not find an answer.

Is it possible to store the Application definitions in git also? Is there a generator that can read and sync this?

Right now, im using SCM generator but i want to add image updater annotations and thats not working out.

So the plan is to manually create the Apps (again) but i was hoping i can so that in git instead of directly in Argo.

Thanks in advance!

**Comment by krancour** [comment, marked as answer]
Possible, and preferable even. The pattern is called "app of apps," so that is the search term that should lead you to many useful results.

**Reply by BeyerJC** [reply]
I saw the "App of Apps" but misunderstood its purpose. Thank you !

**Reply by BeyerJC** [reply]
Just a quick follow up question: Can i use the AppOfApps also to store my ApplicationSets the same way ? I have multiple PullRequest Generators.

When i try to apply it, it gives me an error because of the fields e.g {{ number }} in the template part.

Im using Helm to save the definitions in my git btw. Is there a workaround for that ? Can it be escaped ?

**Reply by BeyerJC** [reply]
UPDATE: It is possible to deploy applicationSets with AppOfApps pattern but you can not use Helm for that. You can create an application pointing directly into a folder containing the appSet definitions.

**Reply by krancour** [reply]
If you need to involve Helm, you might want to investigate the "rendered branches" pattern. You can use whatever tools you like to pre-render everything, so that by the time Argo gets its hands on it, they are just plain manifests.
